### PR TITLE
Documenting based models

### PR DESCRIPTION
**Pull request by williamFalcon**

### CODE REVIEW

File: docs/source/vision_tasks.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+The `1:1` task:
+
+Corresponds to using the feature maps $m_{i} \in M^a$ vs $m_{j} \in M^+$ in the NCE Loss $\math{cal}(m_i, m_j)$
+
+.. code-block:: python
+
+    xa, x+ = augmentations(x), augmentations(x)
+    ra = model(xa)
+    r+ = model(x+)
```
Comment (Borda) [inline docs/source/vision_tasks.rst:30]: is this correct `r+`?

Comment (williamFalcon) [reply docs/source/vision_tasks.rst:30]: rpos?

Comment (williamFalcon) [reply docs/source/vision_tasks.rst:30]: it’s a positive, anchor and negative representation

Comment (Borda) [reply docs/source/vision_tasks.rst:30]: yeas but `+` is not allowed in var name... so `r_pos` ?

Comment (Borda) [reply docs/source/vision_tasks.rst:30]: probably not too important in code but in examples rather longer names so user can easier follow...

File: docs/source/self_supervised_models.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+===============
+
+CPC (V2)
+--------
+PyTorch implementation of `Data-Efficient Image Recognition with Contrastive Predictive Coding <https://arxiv.org/abs/1905.09272>`_
+by (Olivier J. Hénaff, Aravind Srinivas, Jeffrey De Fauw, Ali Razavi, Carl Doersch, S. M. Ali Eslami, Aaron van den Oord).
+
+.. code-block:: python
+
+    from pl_bolts.models.self_supervised import CPCV2
```
Comment (Borda) [inline docs/source/self_supervised_models.rst:11]: as it is v for the version I would rather use lower so it clear that V is not part of the shortcut.. `CPCv2`

File: docs/source/vision_tasks.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+Self-supervised learning in computer vision uses a pretext task to extract learning signal.
+Here is a collection of common tasks
+
+AMDIMContrastiveTask
+^^^^^^^^^^^^^^^^^^^^
+Implementation modified from the `original repo <https://github.com/Philip-Bachman/amdim-public>`_.
+This is the contrastive task from AMDIM (`Philip Bachman, R Devon Hjelm, William Buchwalter <https://arxiv.org/abs/1906.00910>`_).
+
+In this task, we take in two sets of feature maps
+(in this case from a positive $x^+$ and anchor example $x^a$) $M^a = \{m_1, m_2, ..., m_i\}, M^+ = \{m_1, m_2, ..., m_j\}$.
```
Comment (Borda) [inline docs/source/vision_tasks.rst:18]: you mind use the math 
```
.. math::

    M^a = \{m_1, m_2, ..., m_i\}, M^+ = \{m_1, m_2, ..., m_j\}

```

File: docs/source/vision_tasks.rst (hunk 1 above)
Comment (Borda) [reply docs/source/vision_tasks.rst:30]: as this example is no valid code lol

File: docs/source/vision_tasks.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    # 1:1 task (last two feature maps)
+    task = AMDIMContrastiveTask('1:1')
+    map_i = ra[-1]
+    map_j = r+[-1]
+    loss = task(map_i, map_j)
+
+The `1:5,1:7,5:5` task:
+This is the task used in the AMDIM paper. This task generates three losses.
+Corresponds to using the feature maps $m_{i} \in M^a$ vs $m_{j} \in M^+$ in the NCE Loss $\math{cal}(m_i, m_j)$
+
```
Comment (Borda) [inline docs/source/vision_tasks.rst:41]: also add `.. math::`

File: pl_bolts/models/gans/basic/basic_gan_module.py [hunk 5]
```diff
@@ -127,9 +128,9 @@ def training_step(self, batch, batch_idx, optimizer_idx):
             return self.discriminator_step(x)
 
     def configure_optimizers(self):
-        lr = self.learning_rate
-        b1 = self.b1
-        b2 = self.b2
+        lr = self.hparams.learning_rate
```
Comment (Borda) [inline pl_bolts/models/gans/basic/basic_gan_module.py:131]: we shall increase min PL version for this, right?

File: pl_bolts/models/self_supervised/cpc/cpc_module.py [hunk 6]
```diff
@@ -28,18 +28,34 @@
 
 class CPCV2(pl.LightningModule):
 
-    def __init__(self, hparams):
+    def __init__(self,
+                 encoder='cpc_encoder',
```
Comment (Borda) [inline pl_bolts/models/self_supervised/cpc/cpc_module.py:32]: would be nice to have type hints :]

Comment (williamFalcon) [reply pl_bolts/models/self_supervised/cpc/cpc_module.py:32]: this is still a research project lol...

**Pull request merged**